src/utils/facet-utils.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  FacetOption,
+  getDefaultSelectedFacets,
+  type FacetBucket,
+  type SelectedFacets,
+} from '../models';
+
+// This file contains several helper functions designed to make working immutably
+// with SelectedFacet objects and FacetGroups cleaner & easier.
```

Comment:
I split off several helper functions to deal with some common facet-related actions we perform across various components.